spacy/tests/matcher/test_matcher_api.py

```diff
@@ -265,14 +265,25 @@ def test_matcher_regex_shape(en_vocab):
     assert len(matches) == 0
 
 
-def test_matcher_compare_length(en_vocab):
+@pytest.mark.parametrize(
+    "cmp, ngood, bad", 
+    [
+        ("==", 1, "a aaa"),
+        ("!=", 2, "aa"),
+        (">=", 2, "a"),
+        ("<=", 2, "aaa"),
+        (">", 1, "a aa"),
+        ("<", 1, "aa aaa")
+    ]
```

Comment:
I don't think you need `ngood` here: it's always just `3 - length(bad)`, right? Can `bad` be a list (e.g. `[a, aa]`) to make its meaning a bit more clear (and remove the `split()`)

Reply:
I do like how you've generalized this unit test, but it took me a second to understand what was going on :-)

Reply:
Thank you for your feedback, you're definitely right on the optimisation and the ugliness!
I'll fix the test with your hints.